src/directives/leaflet.js

```diff
@@ -24,9 +23,9 @@ angular.module("leaflet-directive", []).directive('leaflet',
         transclude: true,
         template: '<div class="angular-leaflet-map"><div ng-transclude></div></div>',
         controller: function ($scope) {
-            _leafletMap = $q.defer();
+            this._leafletMap = $q.defer();
```

Comment:
Actually, cleaning the var on line 141 was what I did initially. It also works.
However, linking it to the controller object let Angular manage it since the controller is destroyed as well on scope.$destroy.

In my opinion, _leafletMap should be an attribute of the controller since it is only used with it. Being forced to manually delete it on line 141 proves it.